conflation/mapillary.py

```diff
@@ -1,60 +1,106 @@
+    :param traces_source: Dict of configs, see the conf param of make_trace_data_requests()
+    :param processes: Number of threads to use
+    """
     # Do a quick check to see if user specified the mandatory 'client_id' in traces_source JSON
-    if 'client_id' not in traces_source:
-        raise KeyError('Missing "client_id" (Mapillary Client ID) key in --traces-source JSON.')
-
-    # Create dirs
-    output_dir, output_tmp_dir = util.initialize_dirs(bbox)
+    if "client_id" not in traces_source:
+        raise KeyError(
+            'Missing "client_id" (Mapillary Client ID) key in --traces-source JSON.'
+        )
 
     # Break the bbox into sections and save it to a pickle file
     bbox_sections = util.split_bbox(output_dir, bbox, to_bbox)
 
-    finished_bbox_sections = multiprocessing.Value('i', 0)
-    with multiprocessing.Pool(initializer=initialize_multiprocess,
-                              initargs=(output_dir, output_tmp_dir, traces_source, finished_bbox_sections),
-                              processes=processes) as pool:
-        result = pool.map_async(pull_and_save_trace_for_bbox, bbox_sections)
+    finished_bbox_sections = multiprocessing.Value("i", 0)
+    with multiprocessing.Pool(
+        initializer=initialize_multiprocess,
+        initargs=(output_dir, output_tmp_dir, traces_source, finished_bbox_sections),
+        processes=processes,
+    ) as pool:
+        result = pool.map_async(pull_filter_and_save_trace_for_bbox, bbox_sections)
 
-        print('Placing {} results in {}...'.format(len(bbox_sections), output_dir))
+        print("Placing {} results in {}...".format(len(bbox_sections), output_dir))
         progress = 0
         increment = 5
         while not result.ready():
             result.wait(timeout=5)
             next_progress = int(finished_bbox_sections.value / len(bbox_sections) * 100)
             if int(next_progress / increment) > progress:
-                print('Current progress: {}%'.format(next_progress))
+                print("Current progress: {}%".format(next_progress))
                 progress = int(next_progress / increment)
         if progress != 100 / increment:
-            print('Current progress: 100%')
+            print("Current progress: 100%")
 
         # TODO: Delete the tmp dir after run?
 
 
-def to_bbox(llo, lla, mlo, mla):
-    return ','.join([str(llo), str(lla), str(mlo), str(mla)])
+def to_bbox(llo: float, lla: float, mlo: float, mla: float) -> str:
+    """
+    Given (min_lon, min_lat, max_lon, max_lat) bounding box values, returns a string representation understood by
+    Mapillary APIs.
+    """
+    return ",".join([str(llo), str(lla), str(mlo), str(mla)])
+
+
+def is_within_bbox(lon: float, lat: float, bbox: list[float]) -> bool:
+    """
+    Checks if lon / lat coordinate is within a bbox in the format of [min_lon, min_lat, max_lon, max_lat]
+    """
+    return bbox[0] <= lon < bbox[2] and bbox[1] <= lat < bbox[3]
```

Comment:
ive always wished but never knew this syntax sugar existed!

Reply:
It's fairly new! From Python 3.8 I believe